aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -399,48 +417,138 @@ inline Return Dispatcher::callWithDispatchKey(const TypedOperatorHandle<Return(A
       }
     }
     // keeping the guard alive while executing the kernel
-    return kernel.template call<Return, Args...>(op, std::forward<Args>(args)...);
+    return kernel.template call<Return, Args...>(op, DispatchKeySet(dispatchKey), std::forward<Args>(args)...);
```

Comment:
could this break downstream kernels? According to the invariant, they might expect keys below themselves to be set (say a variable kernel might expect the CPU bit to be set) but here you're generating a dispatch key set that would only have the variable bit and no CPU bit if somebody calls callWithDispatchKey(..., DispatchKey::Variable, ...)

Reply:
Yeah you're completely right, thanks! I doesn't break any CI tests currently because that API is no longer used by the core dispatcher logic. I should probably:
- Fix that by having `callWithDispatchKey` call into the dispatchKeyExtractor to grab the correct set, masking appropriately so the passed-in key has highest priority
- Adding a unit test so that this PR would have failed as is- a redispatching test that calls `callWithDispatchKey` (I haven't added any unit tests yet, it's on the list after "sanity-check benchmarks and respond to other major feedback")

Reply:
if callWithDispatchKey calls into the dispatchKeyExtractor to grab the correct set, that'll likely have perf implications and you'll have to remeasure the impact